intelmq/lib/utils.py

```diff
@@ -808,3 +816,44 @@ def file_name_from_response(response: requests.Response) -> str:
     except KeyError:
         file_name = response.url.split("/")[-1]
     return file_name
+
+
+def list_all_bots() -> str:
+    bots = {
+        'Collector': {},
+        'Parser': {},
+        'Expert': {},
+        'Output': {},
+    }
+
+    base_path = resource_filename('intelmq', 'bots')
+
+    botfiles = [botfile for botfile in pathlib.Path(base_path).glob('**/*.py') if botfile.is_file() and botfile.name != '__init__.py']
+    for file in botfiles:
+        file = Path(file.as_posix().replace(base_path, 'intelmq/bots'))
+        mod = importlib.import_module('.'.join(file.with_suffix('').parts))
+        if hasattr(mod, 'BOT'):
+            name = mod.BOT.__name__
+            keys = {}
+            variables = sorted(
+                (key, value) for key, value in vars(mod.BOT).items()
```

Comment:
`vars()` only returns the members of this class, excluding the parent classes. This is problem for the Bot classes and the mixins as well. `dir()`, which I proposed earlier, returns all members - but not a dict, just the names.